connaisseur/image.py

```diff
@@ -27,61 +27,63 @@ class Image:
     tag: Optional[str]
     digest: Optional[str]
 
-    def __init__(self, image: str):
-        separator = r"[-._:@+]|--"
-        alphanum = r"[A-Za-z0-9]+"
-        component = f"{alphanum}(?:(?:{separator}){alphanum})*"
-        ref = f"^{component}(?:/{component})*$"
-
-        # e.g. :v1, :3.7-alpine, @sha256:3e7a89...
-        tag_re = r"(?:(?:@sha256:([a-f0-9]{64}))|(?:\:([\w.-]+)))"
-
-        match = re.search(ref, image)
-        if not match:
-            msg = "{image} is not a valid image reference."
-            raise InvalidImageFormatError(message=msg, image=image)
-
-        name_tag = image.split("/")[-1]
-        search = re.search(tag_re, name_tag)
-        self.digest, self.tag = search.groups() if search else (None, "latest")
-        self.name = name_tag.removesuffix(":" + str(self.tag)).removesuffix(
-            "@sha256:" + str(self.digest)
+    def __init__(self, image: str):  # pylint: disable=too-many-locals
+
+        # implements https://github.com/distribution/distribution/blob/main/reference/regexp.go
+        digest_hex = r"[0-9a-fA-F]{32,}"
```

Comment:
That's for sha256 and not generally digests.

From [spec](https://docs.docker.com/registry/spec/api/#content-digests) this should be more general. Also algorithm and separator do not seem to match the spec

Reply:
Also `docker pull securesystemsengineering/testimage@sha256:2d2c82da46687f6f390389c28f64ad00a21a077e6ded496e2e37e6a15b42682E` gives me `invalid checksum digest format` in comparison to lower-case `e` at the end. Not sure why, whether I looked at the wrong spec or docker does more validation that should be

In the latter case, not sure what to do. Maybe log with info?

Reply:
ok, welp. we pretty much only support `sha256` at this point. it's hardcoded at multiple places in the code and refactoring those would XPLODE the whole PR. so if it's fine with you, i would leave things as the are right now.

Reply:
If you open an issue I'm fine with postponing that.
What about the other issue with format?

Reply:
Created #843. For the format ... I don't konw, I think I'd rather implement the OCI standard (roughly) then try to comply with funky things docker is doing. Agree?